src/Card.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { memo, forwardRef } from "react";
+import type { ReactNode } from "react";
+import { symToStr } from "tsafe/symToStr";
+import { assert } from "tsafe/assert";
+import type { Equals } from "tsafe";
+
+import { FrIconClassName, RiIconClassName } from "./lib/generatedFromCss/classNames";
+import { fr, RegisteredLinkProps } from "./lib";
+import { getLink } from "./lib/routing";
+import { cx } from "./lib/tools/cx";
+
+import "./dsfr/component/card/card.css";
+
+//https://main--ds-gouv.netlify.app/example/component/card/
+export type CardProps = {
+    className?: string;
+    title: ReactNode;
+    linkProps: RegisteredLinkProps;
+    desc?: ReactNode;
+    imageUrl?: string;
+    imageAlt?: string;
+    start?: ReactNode;
+    detail?: ReactNode;
+    end?: ReactNode;
+    endDetail?: ReactNode;
+    badges?: ReactNode[]; // todo: restrict to badge component ? these badges are display on the image
+    footer?: ReactNode; // where actions can be placed
+    horizontal?: boolean;
+    size?: "small" | "medium" | "large"; // only affect the text
+    enlargeLink?: boolean; // make the whole card clickable
+    iconId?: FrIconClassName | RiIconClassName; // only needed when enlargeMode=true
```

Comment:
What is `enlargeMode`?  Wouldn't it make sence to express this in the type def?

Reply:
enlargeMode makes the whole card clickable; what do you mean by `express this in the type def?` ?

Reply:
my bad, its not `enlargeMode` but `enlargeLink`

Reply:
>  enlargeMode makes the whole card clickable; what do you mean by express this in the type def? ?  

```typescript   

type EnlargedLink = {
    enlargeLink: true;
    iconId?: FrIconClassName | RiIconClassName;
};

type NotEnlargedLink = {
    enlargeLink?: false;
    iconId?: never;
}

type Props = .... & (EnlargedLink | NotEnlargedLink);


```

Reply:
Yay makes sense thanks